ldes-server-port-ingest/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/ingest/MemberIngesterImpl.java

```diff
@@ -26,30 +28,29 @@ public MemberIngesterImpl(MemberIngestValidator validator, MemberRepository memb
 		this.eventPublisher = eventPublisher;
 	}
 
-	@Override
-	public void ingest(Member member) {
-		validator.validate(member);
-		final boolean isNewMember = !memberRepository.memberExists(member.getId());
-		final String memberId = member.getId().replaceAll("[\n\r\t]", "_");
-		if (isNewMember) {
-			ingestNewMember(member, memberId);
-		} else {
-			log.warn("Duplicate member ingested. Member with id {} already exist", memberId);
-		}
-	}
-
-	private void ingestNewMember(Member member, String memberId) {
-		Metrics.counter("ldes_server_ingested_members_count").increment();
-		final Member savedMember = save(member);
-		final var memberIngestedEvent = new MemberIngestedEvent(savedMember.getModel(), savedMember.getId(),
-				savedMember.getCollectionName(), savedMember.getSequenceNr());
-		eventPublisher.publishEvent(memberIngestedEvent);
-		log.debug("Member with id {} ingested.", memberId);
-	}
-
-	private Member save(Member member) {
-		member.removeTreeMember();
-		return memberRepository.saveMember(member);
-	}
-
-}
+    @Override
+    public void ingest(Member member) {
+        validator.validate(member);
+        final String memberId = member.getId().replaceAll("[\n\r\t]", "_");
+        ingestNewMember(member, memberId);
+    }
+
+    private void ingestNewMember(Member member, String memberId) {
+        Optional<Member> savedMemberOpt = insert(member);
+        if (savedMemberOpt.isPresent()) {
```

Comment:
Based on that this is the only place we use the return of the method, I think it'll be much better to use a boolean rather than an optional (we lose time mapping it from an db entity to a Member)

Reply:
done, i assumed the mapping had a function (i know assumed)

Reply:
as discussed, mapping is needed, reverted to Optional again